Extend service invocation to non-Dapr endpoints docs (#3384)
dapr/docs: daprdocs/content/en/reference/httpEndpoints-overview.md

> ---

Reviewer: 1) Rename this file to httpendpoint-reference
2) Add another directory level called "Resource specs" including the required "_index.md" file

type: docs
title: "HTTPEndpoint spec"
linkTitle: "HTTPEndpoint spec"

> description: "The HTTPEndpoint Dapr resource spec"

Reviewer: ```suggestion
description: "The HTTPEndpoint resource spec"
```

weight: 300
aliases:
  - "/operations/httpEndpoints/"
---

> The HTTPEndpoint is a Dapr resource that can be used to allow the invocation of non-Daprized endpoints from a Dapr application.

Reviewer: ```suggestion
The HTTPEndpoint is a Dapr resource that is used to enable the invocation of non-Dapr endpoints from a Dapr application.
```

## HTTPEndpoint format

```yaml
apiVersion: dapr.io/v1alpha1
kind: HTTPEndpoint
metadata:
  name: <NAME>  
spec:
  baseUrl: <REPLACE-WITH-BASEURL> # Required. Use "http://" or "https://" prefix.
  headers: # Optional
  - name: <REPLACE-WITH-A-HEADER-NAME>
    value: <REPLACE-WITH-A-HEADER-VALUE>
  - name: <REPLACE-WITH-A-HEADER-NAME>
    secretKeyRef:
      name: <REPLACE-WITH-SECRET-NAME>
      key: <REPLACE-WITH-SECRET-KEY>
scopes: # Optional
  - <REPLACE-WITH-SCOPED-APPIDS>
auth: # Optional
  secretStore: <REPLACE-WITH-SECRETSTORE>
```

## Spec fields

| Field              | Required | Details | Example |
|--------------------|:--------:|---------|---------|

> | baseUrl            | Y        | Base URL of the non-Daprized endpoint  | `"https://docs.dapr.io"`, `"http://api.github.com"`

Reviewer: I would not use docs example here since this is not real. Use an example that has an API that can be called, like the GitHub one. Find another one that is open source and free.

Reviewer: Are any specific validation errors thrown that users should be made aware of?

Author: no specific errors for this resource validation
